### DISCUSSION TITLE
v2.13.27

### DISCUSSION THREAD

**Discussion by danharrin** [filamentphp/filament#3138] [General]
## What's Changed
* CHG: Make filament app.js defer by @bernhardh in https://github.com/filamentphp/filament/pull/3102
* create slug based on path and model class to prevent duplicate slugs by @mvenghaus in https://github.com/filamentphp/filament/pull/2927
* Update turkish translations by @deligoez in https://github.com/filamentphp/filament/pull/3131
* Use Laravel Pint for code style by @zepfietje in https://github.com/filamentphp/filament/pull/3126
* FIX: Respect action align setting by @bernhardh in https://github.com/filamentphp/filament/pull/2952
* Normalize English actions translations by @zepfietje in https://github.com/filamentphp/filament/pull/2885
* Feature: allow storing filters in session by @ralphjsmit in https://github.com/filamentphp/filament/pull/2965
* Add `enableOpen` to FileUpload field by @pxlrbt in https://github.com/filamentphp/filament/pull/2986
* Add Dusk selectors to actions and notification by @tanthammar in https://github.com/filamentphp/filament/pull/3021

## New Contributors
* @mvenghaus made their first contribution in https://github.com/filamentphp/filament/pull/2927
* @deligoez made their first contribution in https://github.com/filamentphp/filament/pull/3131

**Full Changelog**: https://github.com/filamentphp/filament/compare/v2.13.26...v2.13.27

<hr /><em>This discussion was created from the release <a href='https://github.com/filamentphp/filament/releases/tag/v2.13.27'>v2.13.27</a>.</em>

**Comment by xiaomlove** [comment]
In this version, Resource::getSlug() has made a [big change](https://github.com/filamentphp/filament/compare/v2.13.26...v2.13.27#diff-0ac35a26643c2310461ddbcb023f322f67480bd942bf4d5d8cc18de7dd097833), whether to use plural or not is related to locale, so how can I set locale dynamically?

The route is registered when the Service is started, and getSlug() uses the locale in config/app.php, which is en, and returns the plural(`locale_has_pluralization()` will get `true` in `en`). I change the locale in the middleware, for example to zh_CN, so that when rendering the backend menu, getSlug() does not return the plural(because `locale_has_pluralization()` will return `false` in `zh_CN`), resulting in an error occurred that the route is not defined.

 What solution should I use to change the locale before registering the Resource route? I need to dynamically modify the locale based on the authenticated user information, or at least the cookie.

**Reply by danharrin** [reply]
We fixed this in a release today, could you check again?

**Reply by xiaomlove** [reply]
> We fixed this in a release today, could you check again?

Good job, it's OK now, thanks!
